cdc/sink/mysql_params.go

```diff
@@ -287,6 +285,18 @@ func generateDSNByParams(
 		dsnCfg.Params["tidb_txn_mode"] = txnMode
 	}
 
+	// Since we don't need select, just set default isolation level to read-committed
+	// transaction_isolation is mysql newly introduced variable and will vary from MySQL5.7/MySQL8.0/Mariadb
+	isolation, err := checkTiDBVariable(ctx, testDB, "transaction_isolation", defaultTxnIsolationRC)
+	if err != nil {
+		return "", err
+	}
+	if isolation != "" {
+		dsnCfg.Params["transaction_isolation"] = fmt.Sprintf(`"%s"`, defaultTxnIsolationRC)
+	} else {
+		dsnCfg.Params["tx_isolation"] = fmt.Sprintf(`"%s"`, defaultTxnIsolationRC)
+	}
```

Comment:
What about checking and setting, in case of both `transaction_isolation` and `tx_isolation` are unknown system variable.

```go

err = checkAndSet("transaction_isolation")
// check error
err = checkAndSet("tx_isolation)
// check error

func checkAndSet(var string) error {
  isolation, err := checkTiDBVariable(ctx, testDB, var, defaultTxnIsolationRC)
  if err != nil {
    return err
  }
  if isolation != "" {
    dsnCfg.Params[var] = fmt.Sprintf(`"%s"`, defaultTxnIsolationRC)
  }
  return nil
}

```

Reply:
It may happens for some very old mysql.  
But I think it's ok now.

Reply:
well, these two variables co-exist in some MySQL versions.